catalog/MDCCatalog/MDCCatalogDebugAlert.swift

```diff
@@ -0,0 +1,200 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import UIKit
+
+import MaterialComponents.MaterialDialogs
+
+class MDCCatalogDebugSetting {
+  init(title: String) {
+    self.title = title
+  }
+
+  public var title: String
+  public var setter: (Bool) -> Void = { _ in }
+  public var getter: () -> Bool = { false }
+}
+
+class MDCCatalogDebugAlert: UICollectionViewController {
+  fileprivate var settings = [MDCCatalogDebugSetting]()
+
+  fileprivate let layout = UICollectionViewFlowLayout()
+  fileprivate let dialogTransitionController = MDCDialogTransitionController()
+
+  fileprivate var numberOfRows: Int {
+    return settings.count + 1
+  }
+  fileprivate var dismissCellRow: Int {
+    return settings.count
+  }
+
+  override var preferredContentSize: CGSize {
+    get { return CGSize.init(width: 400, height: self.numberOfRows * 44) }
+    set { super.preferredContentSize = newValue }
+  }
+
+  required init?(coder aDecoder: NSCoder) {
```

Comment:
You can mark this API as unavailable for this class.

Reply:
Goes for all of them.

Reply:
Prepended those methods with @available(*, unavailable).